Common/Statistics/FitnessScore.cs

```diff
@@ -0,0 +1,219 @@
+
+                    // just in case...
+                    if (annualFactor == 0)
+                    {
+                        return 0;
+                    }
+
+                    var portfolioAnnualizedReturn = currentPortfolioReturn / annualFactor;
+
+                    var sortinoRatio = GetSortinoRatio(portfolioAnnualizedReturn);
+
+                    var returnOverMaxDrawdown = GetReturnOverMaxDrawdown(currentPortfolioValue, portfolioAnnualizedReturn);
+
+                    var portfolioTurnover = GetPortfolioTurnover(currentPortfolioValue);
+
+                    var rawFitnessScore = portfolioTurnover * (returnOverMaxDrawdown + sortinoRatio);
+
+                    return Math.Round(ScaleToRange(rawFitnessScore, maximumValue: 10, minimumValue: -10), 3);
+                }
+            }
+            catch (Exception exception)
+            {
+                Log.Error(exception);
+            }
+
+            return 0;
+        }
+
+        private decimal GetSortinoRatio(decimal portfolioAnnualizedReturn)
+        {
+            var closedTrades = _algorithm.TradeBuilder.ClosedTrades;
+            if (_previousClosedTradeIndex < closedTrades.Count)
+            {
+                var updateStandardDeviation = false;
+                for (; _previousClosedTradeIndex < closedTrades.Count; _previousClosedTradeIndex++)
+                {
+                    if (closedTrades[_previousClosedTradeIndex].ProfitLoss < 0)
+                    {
+                        updateStandardDeviation = true;
+                        _losses.Add((double)closedTrades[_previousClosedTradeIndex].ProfitLoss);
+                    }
+                }
+
+                // we update the profit loss downside deviation only when a new loss trade was closed
+                if (updateStandardDeviation)
+                {
+                    var variance = _losses.Variance();
+                    _profitLossDownsideDeviation = Math.Sqrt(variance);
+                }
+            }
+
+            var sortinoRatio = decimal.MaxValue;
+            if (_losses.Count > 1)
+            {
+                sortinoRatio = (portfolioAnnualizedReturn - _riskFreeRate) / (decimal)_profitLossDownsideDeviation;
+            }
+
+            return SigmoidalScale(sortinoRatio);
+        }
+
+        private decimal GetReturnOverMaxDrawdown(decimal currentPortfolioValue, decimal portfolioAnnualizedReturn)
+        {
+            if (currentPortfolioValue > _maxPortfolioValue)
+            {
+                _maxPortfolioValue = currentPortfolioValue;
+            }
+
+            var currentDrawdown = currentPortfolioValue / _maxPortfolioValue - 1;
+
+            _maxDrawdown = currentDrawdown < _maxDrawdown ? currentDrawdown : _maxDrawdown;
+
+            var returnOverMaxDrawdown = decimal.MaxValue;
+            if (_maxDrawdown != 0)
+            {
+                returnOverMaxDrawdown = portfolioAnnualizedReturn / Math.Abs(_maxDrawdown);
```

Comment:
Do you need to use CAGR here? (not a simple `totalReturn/years`). e.g. with a 20 year backtest, 10% return per year, it would look like 33% return per year and 3.3 RoMaD (assuming 10% DD). But 10% annual return/10% DD should be 1 RoMaD. 

This actually sounds more like the MAR/Calmar ratio: https://en.wikipedia.org/wiki/Calmar_ratio
"the MAR Ratio, equal to the compound annual return from inception, divided by the maximum drawdown from inception."

Worth further discussion https://www.quantconnect.com/forum/discussion/5888/alpha-stream-scoring-criteria/p1

Reply:
Nice catch! Updated 👍